File: 백준/Silver/1920. 수 찾기/수 찾기.py
def finalanswer(li1, li2):
    # li2를 집합으로 변환하여 빠른 검색을 가능하게 합니다.
    set_li1 = set(li1)

    # 결과를 저장할 리스트 생성
    li3 = []

    # li1의 각 요소를 순회
    for item in li2:
        # item이 set_li2에 존재하면 1, 그렇지 않으면 0을 추가합니다.
        if item in set_li1:
            li3.append(1)
        else:
            li3.append(0)

    return li3

# 첫 번째로 정수의 개수를 입력받기
F = int(input())

# 정수를 공백으로 구분하여 입력받기
Fnum = input()

# 입력된 문자열을 공백으로 나누어 리스트로 변환
li1 = list(map(int, Fnum.split()))

# 검수받을 수 갯수
R = int(input())

# 정수를 공백으로 구분하여 입력받기
Rnum = input()

li2 = list(map(int, Rnum.split()))

# 결과를 리스트로 얻기
result = finalanswer(li1, li2)

# 결과 리스트를 엔터로 구분하여 출력
for item in result:
    print(item)

File: 백준/Silver/1920. 수 찾기/수 찾기.java
import java.io.*;
import java.util.*;

public class Main {

    // 이진탐색
    public static List<Integer> finalAnswer(int[] li1, int[] li2) {
        List<Integer> li3 = new ArrayList<>();
        Arrays.sort(li1); // li1 정렬
        
        for (int target : li2) {
            int left = 0, right = li1.length - 1;
            boolean found = false;

            while (left <= right) {
                int mid = (left + right) / 2;
                if (li1[mid] == target) {
                    li3.add(1); //성공 시
                    found = true;
                    break;
                } else if (li1[mid] > target) {
                    right = mid - 1;
                } else {
                    left = mid + 1;
                }
            }

            if (!found) li3.add(0); //실패 시
        }

        return li3;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;


        int F = Integer.parseInt(br.readLine());
        int[] li1 = new int[F];
        
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < F; i++) {
            li1[i] = Integer.parseInt(st.nextToken());
        }


        int R = Integer.parseInt(br.readLine());
        int[] li2 = new int[R];
        
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < R; i++) {
            li2[i] = Integer.parseInt(st.nextToken());
        }

        List<Integer> result = finalAnswer(li1, li2);

        //출력
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        for (int item : result) {
            bw.write(item + "\n");
        }
        bw.flush();
        bw.close();
        br.close();
    }
}
